ops/model.py

```diff
@@ -60,6 +62,16 @@ def get_unit(self, unit_name):
     def get_app(self, app_name):
         return self._cache.get(Application, app_name)
 
+    @property
+    def bindings(self):
+        if self._bindings is None:
+            binding_keys = list(self._meta.relations) + list(self._meta.extra_bindings)
+            for endpoint_relations in self.relations.values():
+                if endpoint_relations:
+                    binding_keys.extend(endpoint_relations)
+            self._bindings = BindingMapping(binding_keys, self._backend)
```

Comment:
Why is this diverging from the other patterns we have?

Reply:
The problem I tried to address here is that the network info represented by `Binding` objects may be different depending on whether just a `binding_name` is present or both a `binding_name` and `relation_id` are present.

Scenarios:

1. no relations are created yet, `network-get <binding-name>` will return a valid network info not associated with a particular `relation_id`;
2. one or multiple non-CMR relations are created, both `network-get <binding-name>` and  `network-get -r <any-relation-id> <binding-name>`, where `<any-relation-id>` is any relation ID associated with `relation_name == <binding-name>`;
3. A non-CMR and a CMR relations exist. `network-get <binding-name>` and `network-get -r <non-CMR-rid>` will return the same content while `network-get -r <CMR-rid> <binding-name>` may return a different `ingress-address`, see https://github.com/juju/juju/blob/juju-2.6.7/state/relationunit.go#L634-L658.

I could try and handle it the same way as we do for relations based on our discussion for #107:

```
    def __getitem__(self, binding_name):
        binding_list = self._data[binding_name]
        if binding_list is None:
            binding_list = self._data[binding_name] = []
            binding_list.append(Binding(binding_name), None, backend))
            for relation_id in self._backend.relation_ids(binding_name):
                binding_list.append(Binding(binding_name, relation_id, backend))
        return relation_list

   def _get_unique(self, binding_name, relation_id):
        # ...
```

Reply:
Addressed in https://github.com/canonical/operator/pull/109/commits/1dbb9d7c6ba68ecc7c726869c64b4bad5a48bd13